in_toto/verifylib.py

```diff
@@ -306,294 +308,582 @@ def verify_all_steps_command_alignment(layout, links_dict):
+def verify_create_rule(rule, source_materials_queue, source_products_queue):
   """
   <Purpose>
-    Verifies that path pattern - 2nd element of rule - matches at least one
-    file in the artifact queue. This might conflict with common understanding of
-    glob patterns (especially "*").
+    The create rule guarantees that no product filtered by the pattern, already
+    appears in the materials queue, i.e. that it was created in that step.
+
+  <Notes>
+    The create rule always passes if the pattern does not match any products:
+
+    rule: ["CREATE", "*"]
+    source materials queue: ["foo"], source products queue: []
+    PASS (makes sense?)
+
+    rule: ["CREATE", "foo"]
+    source materials queue: ["foo"], source products queue: []
+    PASS (might seem strange)
 
     The CREATE rule DOES NOT verify if the artifact has appeared in previous or
     will appear in later steps of the software supply chain.
 
   <Arguments>
     rule:
-            The rule to be verified. Format is ["CREATE", "<path pattern>"]
+            ["CREATE", "<path pattern>"]
+            See https://docs.python.org/2/library/fnmatch.html for wildcards
 
-    artifact_queue:
-            A list of artifact paths that were not matched by a previous rule.
+    source_materials_queue:
+            A list of material paths that were not matched by a previous rule.
 
-  <Exceptions>
-    raises an FormatError if the rule does not conform with the rule format.
-    TBA (see https://github.com/in-toto/in-toto/issues/6)
+    source_products_queue:
+            A list of product paths that were not matched by a previous rule.
 
-    RuleVerficationError if nothing is matched in the artifact queue.
+  <Exceptions>
+    RuleVerficationError
+        if a product filtered by the pattern also appears in the materials
+        queue.
 
   <Side Effects>
-    Uses fnmatch.filter which translates a glob pattern to re.
+    None.
 
   <Returns>
-    The artifact queue minus the files that were matched by the rule.
+    The updated products queue (minus newly created artifacts).
 
   """
-  check_matchrule_syntax(rule)
-  path_pattern = rule[1]
-  matched_artifacts = fnmatch.filter(artifact_queue, path_pattern)
-  if not matched_artifacts:
-    raise RuleVerficationError("Rule {0} failed, no artifacts were created"
-        .format(rule))
+  rule_data = in_toto.artifact_rules.unpack_rule(rule)
 
-  return list(set(artifact_queue) - set(matched_artifacts))
 
+  matched_products = fnmatch.filter(
+      source_products_queue, rule_data["pattern"])
 
-def verify_delete_rule(rule, artifact_queue):
+  for matched_product in matched_products:
+    if matched_product in source_materials_queue:
+      raise RuleVerficationError("Rule '{0}' failed, product '{1}' was found"
+          " in materials but should have been newly created."
+          .format(rule, matched_product))
+
+  return list(set(source_products_queue) - set(matched_products))
```

Comment:
I'm not married to having the set difference operation, especially as we need to temporarily cast. Using pip would require an additional loop over the list, though. I think we can leave it for now.